pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -83,35 +85,82 @@ func writeVifFile(buf []byte, pid, name string) error {
 }
 
 func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, uid string) error {
+	cache := PodCacheInterface{Iface: iface}
+
+	ipv4, ipv6, err := readIPAddressesFromLink(podInterfaceName)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case ipv4 != "" && ipv6 != "":
+		cache.PodIPs, err = sortIPsBasedOnPrimaryIP(ipv4, ipv6)
+		if err != nil {
+			return err
+		}
+	case ipv4 != "":
+		cache.PodIPs = []string{ipv4}
+	case ipv6 != "":
+		cache.PodIPs = []string{ipv6}
+	default:
+		return nil
+	}
+
+	cache.PodIP = cache.PodIPs[0]
+	err = writeToCachedFile(cache, util.VMIInterfacepath, uid, iface.Name)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to write pod Interface to cache, %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func readIPAddressesFromLink(podInterfaceName string) (string, string, error) {
 	link, err := Handler.LinkByName(podInterfaceName)
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
-		return err
+		return "", "", err
 	}
+
 	// get IP address
 	addrList, err := Handler.AddrList(link, netlink.FAMILY_ALL)
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to get a address for interface: %s", podInterfaceName)
-		return err
+		return "", "", err
 	}
+
 	// no ip assigned. ipam disabled
 	if len(addrList) == 0 {
-		return nil
+		return "", "", nil
 	}
 
+	var ipv4, ipv6 string
 	for _, addr := range addrList {
 		if addr.IP.IsGlobalUnicast() {
-			err = writeToCachedFile(PodCacheInterface{Iface: iface, PodIP: addr.IP.String()},
-				util.VMIInterfacepath, uid, iface.Name)
-			if err != nil {
-				log.Log.Reason(err).Errorf("failed to write pod Interface to cache, %s", err.Error())
-				return err
+			if netutils.IsIPv6(addr.IP) && ipv6 == "" {
+				ipv6 = addr.IP.String()
+			} else if !netutils.IsIPv6(addr.IP) && ipv4 == "" {
+				ipv4 = addr.IP.String()
 			}
-			return nil
 		}
 	}
 
-	return nil
+	return ipv4, ipv6, nil
+}
+
+func sortIPsBasedOnPrimaryIP(ipv4, ipv6 string) ([]string, error) {
+	ipv4Primary, err := Handler.IsIpv4Primary()
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to invoke IsIpv4Primary")
```

Comment:
I see no reason to log here anything. There is only one reason for a failure based on `IsIpv4Primary`, and it is recorded in the error itself. Mentioning that we failed to invoke something does not help.
I see that the error is returned back, so I suggest to print it only when it is handled and not in the middle (like you do with all the others).

Reply:
done, both

when there will be a lgtm i will squash all

Reply:
Thank you.
I see you moved the logging to one later up, but on that layer there is also no handling of the error (by handling I mean any operation that is not trowing back the error to upper layers).
If I try to trace back where something is logged, I see the error is wrapped with info only at `processVmUpdate` and then returned to upper layers again.
Bottom line, I see no reason to have anything printed here, the error will pop up until handled.

Reply:
Regarding the squash, I can only LGTM a proper PR, which includes structured commits. But I can informally say so as a comment so you can proceed then.

Reply:
done, and squashed, no need formally, thx.
just done that commits so its easier to review of course.